modules/fedimint-mint-client/src/backup.rs

```diff
@@ -0,0 +1,69 @@
+use fedimint_client::backup::Metadata;
+use fedimint_core::api::{DynFederationApi, GlobalFederationApi};
+use fedimint_core::db::ModuleDatabaseTransaction;
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::{Tiered, TieredMulti};
+use serde::{Deserialize, Serialize};
+
+use super::MintClientModule;
+use crate::{NoteIndex, SpendableNote};
+
+/// Snapshot of a ecash state (notes)
+///
+/// Used to speed up and improve privacy of ecash recovery,
+/// by avoiding scanning the whole history.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Encodable, Decodable)]
+pub struct EcashBackup {
+    notes: TieredMulti<SpendableNote>,
+    pending_notes: Vec<(OutputFinalizationKey, NoteIssuanceRequests)>,
+    epoch_count: u64,
+    next_note_idx: Tiered<NoteIndex>,
+    metadata: Metadata,
+}
+
+impl EcashBackup {
+    /// An empty backup with, like a one created by a newly created client.
+    fn new_empty() -> Self {
+        Self {
+            notes: TieredMulti::default(),
+            pending_notes: vec![],
+            epoch_count: 0,
+            next_note_idx: Tiered::default(),
+            metadata: Metadata::empty(),
+        }
+    }
+}
+
+impl MintClientModule {
+    async fn prepare_plaintext_ecash_backup(
+        &self,
+        dbtx: &mut ModuleDatabaseTransaction<'_>,
+        api: DynFederationApi,
+        metadata: Metadata,
+    ) -> anyhow::Result<EcashBackup> {
+        // fetch consensus height first - so we dont miss anything when scanning
+        let epoch_count = api.fetch_epoch_count().await?;
+
+        let notes = Self::get_all_spendable_notes(&mut dbtx).await;
+
+        let pending_notes: Vec<_> = dbtx
```

Comment:
How do I get these / how are they named&represented in the new client world? :)

Reply:
We discussed that on a call, just backing up all the output `Created` and  states should suffice.